pkg/objectstore/directory.go

```diff
@@ -61,9 +61,22 @@ func (d *directory) GetDirectory(ctx context.Context, dir string) (Directory, er
 		return d, nil
 	}
 	dir = d.absDirName(dir)
-	_, err := d.bucket.container.Item(cloudName(dir))
-	if err != nil {
+	if _, err := d.bucket.container.Item(cloudName(dir)); err == nil {
+		return &directory{
+			bucket: d.bucket,
+			path:   dir,
+		}, nil
+	}
+
```

Comment:
[minor] you could check for a 404 here and return the error otherwise.

Reply:
Makes sense - will follow up. I think I might need to account for error differences b/w GCS, S3, etc - so would need some refactoring to do that.